### PR TITLE
Set current page from URL param page

### LINKED ISSUE #2388: Page number not deeplinking in Discover app
https://incidentdatabase.ai/apps/discover/?page=2

Loads page 1.

### PR DESCRIPTION
**Pull request by clari182**
- closes #2388 

URL: https://pr-2460--staging-aiid.netlify.app/

### CODE REVIEW

**Review by pdcp1 — APPROVED** [review]
LGTM 👍

**Comment by cesarvarela** [comment]
I'll would like to merge this after #2467

File: site/gatsby-site/src/components/discover/createURL.js
Comment (kepae) [inline site/gatsby-site/src/components/discover/createURL.js:46]: This seems like a bad idea -- we don't typecheck or check the return value of this function where it's used:

https://github.com/responsible-ai-collaborative/aiid/blob/1caec141935186372ed9c239951405a4a681b326/site/gatsby-site/src/components/discover/createURL.js#L101

If we want to return a null, we should probably just return null and handle it as an error at the callsite. What do you think?

(apologies for the drive-by comment -- looking at this since we should merge it soon!)

**Comment by cesarvarela** [comment]
We will merge once the tests pass. Please wait for #2592 to be merged and the merge `staging` here.

**Comment by kepae** [comment]
@clari182 @cesarvarela looks good to update this with latest staging, test, and merge?

**Comment by clari182** [comment]
@cesarvarela ready for review

**Review by cesarvarela — APPROVED** [review]
I'm not a fan of the implementation but everything seems to be working so :shipit: 

@kepae

**Pull request merged**
